source/SchemeEditor/SchemeGraphics.cpp

```diff
@@ -121,12 +138,14 @@ void SchemeGraphics::addLevel(Level level)
   LevelItem *levrend = new LevelItem(level, LevelItem::DaughterLevelType,
                                      parentpos_, visual_settings_, scene_);
   connectItem(levrend);
+  delete daughter_levels_[level.energy()];
```

Comment:
should we check to make sure it's non-null before deleting it?

Reply:
Thanks for the review!
if it's null, `delete nullptr` is a no-op, so it's not really necessary to check for it?

Reply:
Any way to use RAII here to avoid having deletes alltogether? I don't recall how the Qt patterns might constrain this.